src/StepAdaptationController/include/WalkingControllers/StepAdaptationController/StepAdaptationController.hpp

```diff
@@ -0,0 +1,272 @@
+/**
+ * @file StepAdaptationController.hpp
+ * @authors Milad Shafiee <milad.shafiee@iit.it>
+ * @copyright 2020 iCub Facility - Istituto Italiano di Tecnologia
+ *            Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ * @date 2020
+ */
+
+#ifndef WALKING_CONTROLLERS_STEP_ADAPTATION_CONTROLLERS_H
+#define WALKING_CONTROLLERS_STEP_ADAPTATION_CONTROLLERS_H
+
+
+// std
+#include <memory>
+
+// iDynTree
+#include <iDynTree/Core/Triplets.h>
+#include <iDynTree/Core/SparseMatrix.h>
+#include <iDynTree/ConvexHullHelpers.h>
+#include <iDynTree/Core/VectorDynSize.h>
+#include <iDynTree/Core/MatrixDynSize.h>
+#include <iDynTree/Core/Twist.h>
+#include <iDynTree/Core/SpatialAcc.h>
+
+// osqp-eigen
+#include <OsqpEigen/OsqpEigen.h>
+
+// yarp
+#include <yarp/os/Value.h>
+
+#include <unordered_map>
+
+#include <qpOASES.hpp>
+#include <WalkingControllers/iDynTreeUtilities/Helper.h>
+
+typedef struct{
```

Comment:
In c++ you can simply do 
```
struct  footTrajectoryGenerationInput {

};
```
http://www.cplusplus.com/doc/tutorial/structures/

Reply:
Thanks. Done.